### PR TITLE
fix get the openapi interface that contains namespace information for deleted items

### LINKED ISSUE #4593: openapi获取某个Namespace信息接口中包含删除的item
- [ ] I have checked the [discussions](https://github.com/ctripcorp/apollo/discussions)
- [ ] I have searched the [issues](https://github.com/ctripcorp/apollo/issues) of this repository and believe that this is not a duplicate.
- [ ] I have checked the [FAQ](https://www.apolloconfig.com/#/zh/faq/common-issues-in-deployment-and-development-phase) of this repository and believe that this is not a duplicate.

**Describe the bug**

**To Reproduce**
1. 
2. 
3. 
4. 

**Expected behavior**
接口返回的items中不包含已删除的item
或在item中标识已删除的状态

**Screenshots**

### Additional Details & Logs

- Version
- Error logs
- Configuration
- Platform and Operating System

### PR DESCRIPTION
**Pull request by CalebZYC**
## What's the purpose of this PR
fix get the openapi interface that contains namespace information for deleted items.

## Which issue(s) this PR fixes:
Fixes #4593

## Brief changelog
fix get the openapi interface that contains namespace information for deleted items

Follow this checklist to help us incorporate your contribution quickly and easily:

- [x] Read the [Contributing Guide](https://github.com/ctripcorp/apollo/blob/master/CONTRIBUTING.md) before making this pull request.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Write necessary unit tests to verify the code.
- [x] Run `mvn clean test` to make sure this pull request doesn't break anything.
- [x] Update the [`CHANGES` log](https://github.com/ctripcorp/apollo/blob/master/CHANGES.md).

### CODE REVIEW

**Comment by nobodyiam** [comment]
There are many places that consume the result of `com.ctrip.framework.apollo.portal.service.NamespaceService#transformNamespace2BO`, e.g. `com.ctrip.framework.apollo.openapi.server.service.ServerNamespaceOpenApiService` and `com.ctrip.framework.apollo.portal.controller.ConfigsExportController`.
However, only the controllers for the namespace UI need the deleted items, e.g. `com.ctrip.framework.apollo.portal.controller.NamespaceController`.

One way to fix the issue is to filter the deleted items in places that we don't need them, or we may add a flag and let the caller decide whether to include the deleted items or not, e.g. `private NamespaceBO transformNamespace2BO(Env env, NamespaceDTO namespace, boolean includeDeletedItems)`.
I think the latter might be better. What do you think?

**Comment by CalebZYC** [comment]
> There are many places that consume the result of `com.ctrip.framework.apollo.portal.service.NamespaceService#transformNamespace2BO`, e.g. `com.ctrip.framework.apollo.openapi.server.service.ServerNamespaceOpenApiService` and `com.ctrip.framework.apollo.portal.controller.ConfigsExportController`. However, only the controllers for the namespace UI need the deleted items, e.g. `com.ctrip.framework.apollo.portal.controller.NamespaceController`.
> 
> One way to fix the issue is to filter the deleted items in places that we don't need them, or we may add a flag and let the caller decide whether to include the deleted items or not, e.g. `private NamespaceBO transformNamespace2BO(Env env, NamespaceDTO namespace, boolean includeDeletedItems)`. I think the latter might be better. What do you think?
I agree with your opinion, but I found that the export namespace config file contains deleted items, this' right?

**Comment by nobodyiam** [comment]
> I agree with your opinion, but I found that the export namespace config file contains deleted items, this' right?

I think the export namespace config file should not contain deleted items, which means we also need to add the `includeDeletedItems` flag to `com.ctrip.framework.apollo.portal.service.NamespaceService#findNamespaceBOs`.

**Review by mghio — CHANGES_REQUESTED** [review]
Please add some unit test in [NamespaceServiceTest](https://github.com/apolloconfig/apollo/blob/master/apollo-portal/src/test/java/com/ctrip/framework/apollo/portal/service/NamespaceServiceTest.java).

File: apollo-portal/src/test/java/com/ctrip/framework/apollo/portal/service/NamespaceServiceTest.java [hunk 1]
```diff
@@ -223,6 +224,43 @@ public void testDeleteEmptyNamespace() {
...
+    List<ItemDTO> itemDTOList = Lists.newArrayList();
+    ItemDTO itemDTO1 = new ItemDTO();
+    itemDTO1.setId(1);
+    itemDTO1.setNamespaceId(1);
+    itemDTO1.setKey("k1");
+    itemDTO1.setValue(String.valueOf(1));
+    itemDTOList.add(itemDTO1);
+
+    ItemDTO itemDTO2 = new ItemDTO();
+    itemDTO2.setId(1);
```
Comment (mghio) [inline apollo-portal/src/test/java/com/ctrip/framework/apollo/portal/service/NamespaceServiceTest.java:246]:
```suggestion
    itemDTO2.setId(2);
```
I think use `itemDTO2.setId(2);` is better. What do you think?

**Comment by nobodyiam** [comment]
I think `com.ctrip.framework.apollo.openapi.server.service.ServerNamespaceOpenApiService#getNamespaces` should also exclude deleted items?

File: apollo-portal/src/test/java/com/ctrip/framework/apollo/portal/service/NamespaceServiceTest.java (hunk 1 above)
Comment (CalebZYC) [reply apollo-portal/src/test/java/com/ctrip/framework/apollo/portal/service/NamespaceServiceTest.java]: It's done

File: apollo-portal/src/test/java/com/ctrip/framework/apollo/portal/service/NamespaceServiceTest.java [hunk 2]
```diff
@@ -223,6 +224,43 @@ public void testDeleteEmptyNamespace() {
...
+
+    List<ItemDTO> deletedItemDTOList = Lists.newArrayList();
+    ItemDTO deletedItemDTO = new ItemDTO();
+    deletedItemDTO.setId(3);
+    deletedItemDTO.setNamespaceId(3);
+    deletedItemDTO.setKey("k3");
+    deletedItemDTOList.add(deletedItemDTO);
+    when(itemService.findDeletedItems(any(), any(), any(), any())).thenReturn(deletedItemDTOList);
+
+    NamespaceBO namespaceBO = namespaceService.loadNamespaceBO(testAppId, testEnv, testClusterName, testNamespaceName);
```
Comment (nobodyiam) [inline apollo-portal/src/test/java/com/ctrip/framework/apollo/portal/service/NamespaceServiceTest.java:261]: It's better to add a case to test `namespaceService.loadNamespaceBO(testAppId, testEnv, testClusterName, testNamespaceName, false);`

Comment (CalebZYC) [reply apollo-portal/src/test/java/com/ctrip/framework/apollo/portal/service/NamespaceServiceTest.java]: > It's better to add a verification to test the result of `namespaceService.loadNamespaceBO(testAppId, testEnv, testClusterName, testNamespaceName, false);` here.

It's done.

**Review by nobodyiam — APPROVED** [review]
LGTM

**Pull request merged**